packages/wallets/src/evm/wallets/device-wallet.ts

```diff
@@ -39,226 +42,416 @@ export class DeviceBrowserWallet extends AbstractClientWallet<
+  /**
+   * creates a new random wallet
+   * @returns the address of the newly created wallet
+   */
+  async generate() {
+    if (this.#ethersWallet) {
+      throw new Error("wallet is already initialized");
     }
-
-    return this.#walletImpl.getWalletData();
+    this.#ethersWallet = Wallet.createRandom();
+    return this.#ethersWallet.address;
   }
 
-  static getAddressStorageKey() {
-    return STORAGE_KEY_ADDR;
-  }
+  /**
+   * create device wallet from the given "encryptedJson" or "privateKey" or "mnemonic"
+   * @param options
+   * @returns
+   */
+  async load(options: LoadOptions) {
+    if (this.#ethersWallet) {
+      throw new Error("wallet is already initialized");
+    }
 
-  static getDataStorageKey() {
-    return STORAGE_KEY_DATA;
-  }
-}
+    if ("encryptedJson" in options) {
+      this.#ethersWallet = await Wallet.fromEncryptedJson(
+        options.encryptedJson,
+        options.password,
+      );
+      return this.#ethersWallet.address;
+    }
 
-export class DeviceWalletImpl extends AbstractWallet {
-  static async fromAsyncStorage(storage: AsyncStorage) {
-    return new DeviceWalletImpl({
-      storage: new AsyncWalletStorage(storage),
-    });
-  }
+    if ("privateKey" in options) {
+      if (!options.encryption && !isValidPrivateKey(options.privateKey)) {
+        throw new Error("invalid private key");
+      }
 
-  static async fromCredentialStore() {
-    return new DeviceWalletImpl({
-      storage: new CredentialsStorage(navigator.credentials),
-    });
-  }
+      const privateKey = await getDecryptor(options.encryption)(
+        options.privateKey,
+      );
+
+      if (
+        options.encryption &&
+        (privateKey === "" || !isValidPrivateKey(privateKey))
+      ) {
+        throw new Error("invalid password");
+      }
+
+      this.#ethersWallet = new Wallet(privateKey);
+      return this.#ethersWallet.address;
+    }
 
-  private options: DeviceWalletImplOptions;
-  #wallet?: ethers.Wallet;
+    if ("mnemonic" in options) {
+      if (!options.encryption && !utils.isValidMnemonic(options.mnemonic)) {
+        throw new Error("invalid mnemonic");
+      }
+
+      const mnemonic = await getDecryptor(options.encryption)(options.mnemonic);
+
+      if (
+        options.encryption &&
+        (mnemonic === "" || !utils.isValidMnemonic(mnemonic))
+      ) {
+        throw new Error("invalid password");
+      }
 
-  constructor(options: DeviceWalletImplOptions) {
-    super();
-    this.options = options;
+      this.#ethersWallet = Wallet.fromMnemonic(mnemonic);
+      return this.#ethersWallet.address;
+    }
   }
 
-  async getSigner(
-    provider?: ethers.providers.Provider,
-  ): Promise<ethers.Signer> {
-    if (!this.#wallet) {
-      throw new Error("Wallet not initialized");
+  /**
+   * initialize the wallet from saved data on storage
+   * @param password - password used for encrypting the wallet
+   */
+  async loadFromStorage(options: LoadFromStorageOptions) {
```

Comment:
let's go with `load` here and `import` instead of your current `load`

so the 4 functions are
- save/load (local usage)
- export/import (external usage)

Reply:
done